internal/pkg/api/server.go

```diff
@@ -102,7 +103,11 @@ func (s *server) Run(ctx context.Context) error {
 	defer func() {
 		err := ln.Close()
 		if err != nil {
-			log.Error().Err(err).Msg("error while closing listener.")
+			if strings.Contains(err.Error(), "use of closed network connection") {
+				log.Warn().Err(err).Msg("server.Run: error while closing listener, closed network connection.")
+			} else {
+				log.Error().Err(err).Msg("server.Run: error while closing listener.")
```

Comment:
sonar check failed, any concerns if we turn these errors down to `Warn` without condition?